cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -106,11 +107,18 @@ def pytest_generate_tests(metafunc):
     for index, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
         vma_data = args['provider'].data.vm_analysis_new
-        vms = vma_data.vms
-        for vm_analysis_key in vms:
-            # Set VM name here
-            new_idlist.append('{}-{}'.format(idlist[index], vm_analysis_key))
-            new_argvalues.append([args["provider"], vm_analysis_key])
+        if 'vms' in vma_data:
+            vms = vma_data.vms
+            for vm_analysis_key in vms:
+                # Set VM name here
+                new_idlist.append('{}-{}'.format(idlist[index], vm_analysis_key))
+                new_argvalues.append([args["provider"], vm_analysis_key])
+        else:
+            # TODO FIXME
+            logger.error("Provider {p} does not have the correct keys in the yaml"
```

Comment:
When using logger, use `%s` in the string and pass the value to be formatted as a 2nd argument.

Reply:
@mshriver : can you please tell why to use %s in logger

Reply:
This is a build-in facility in the python logger that we're taking advantage of, instead of making our own calls to format().

https://docs.python.org/2/library/logging.html#logging.Logger.debug

Reply:
:+1: